### PR TITLE
Added facility related examples

### PR DESCRIPTION
**Pull request by liskaj**
list-levels.js - prints levels
list-rooms.js - prints rooms
list-element-rooms.js - prints elements and related rooms
list-level-assets.js - prints levels and related assets
list-level-rooms.js - prints levels and related rooms

### CODE REVIEW

File: common/tandemClient.js [hunk 1]
```diff
@@ -73,6 +73,37 @@ export class TandemClient {
...
+            headers: {
+                'Authorization': `Bearer ${token}`
+            },
+            body: JSON.stringify(inputs)
+        });
+        const data = await response.json();
+        const results = [];
+
+        for (const item of data) {
+            if ((item[QC.ElementFlags] & ElementFlags.Level) === ElementFlags.Level) {
```
Comment (bakharal) [inline common/tandemClient.js]: I do not think this is a precise check. Usually you just want to check that `item[QC.ElementFlags] & ElementFlags.Level !== 0`

File: common/tandemClient.js [hunk 2]
```diff
@@ -91,6 +122,37 @@ export class TandemClient {
...
+            headers: {
+                'Authorization': `Bearer ${token}`
+            },
+            body: JSON.stringify(inputs)
+        });
+        const data = await response.json();
+        const results = [];
+
+        for (const item of data) {
+            if ((item[QC.ElementFlags] & ElementFlags.Room) === ElementFlags.Room) {
```
Comment (bakharal) [inline common/tandemClient.js]: same here

File: common/utils.js [hunk 3]
```diff
@@ -63,6 +66,33 @@ export class Encoding {
...
+        const result = [];
+        let offset = 0;
+
+        while (offset < binData.length) {
+            const size = binData.length - offset;
+
+            if (size < kElementIdSize) {
+                break;
+            }
+            var elementBuff = Buffer.alloc(kElementIdWithFlagsSize);
```
Comment (bakharal) [inline common/utils.js]: Would be more efficient to create a buffer once and reuse in iterations.

File: facility/list-element-rooms.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    // STEP 3 - iterate through facility models and get all elements from them model
+    for (const link of facility.links) {
+        const elements = await client.getElements(link.modelId, undefined, [ ColumnFamilies.Standard, ColumnFamilies.Refs ]);
+        const roomIndexMap = {};
+
+        // STEP 4 - remember indexes of room elements
+        for (let i = 0; i < elements.length; i++) {
+            const element = elements[i];
+
+            if ((element[QC.ElementFlags] & ElementFlags.Room) === ElementFlags.Room) {
```
Comment (bakharal) [inline facility/list-element-rooms.js]: same problem with flags matching

File: facility/list-level-assets.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
+/*
+    This example demonstrates how to list all levels from facility and find assets for each level.
+    
+    It uses 2-legged authentication - this requires that application is added to facility as service.
```
Comment (bakharal) [inline facility/list-level-assets.js:4]: Do we have this process documented somewhere?

**Review by bakharal — CHANGES_REQUESTED** [review]

File: common/utils.js (hunk 3 above)
Comment (liskaj) [reply common/utils.js]: Fixed

File: facility/list-level-assets.js (hunk 5 above)
Comment (liskaj) [reply facility/list-level-assets.js:4]: Yes, it's documented in readme + we also add details to online doc.

**Review by bakharal — APPROVED** [review]

**Pull request merged**